airflow-core/src/airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -389,7 +389,12 @@ def ti_update_state(
             updated_state=updated_state,
             dag_id=dag_id,
         )
-    except Exception:
+    except Exception as err:
+        if isinstance(err, HTTPException) and err.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY:
+            # We need to rebuild the query and remove  "reschedule_date"which causes DB error
+            data = ti_patch_payload.model_dump(exclude={"reschedule_date"}, exclude_unset=True)
+            query = update(TI).where(TI.id == ti_id_str).values(data)
```

Comment:
Weird we need this… can `_create_ti_state_update_query_and_update_state` not just return the correct query?

Reply:
updated